### PR TITLE
When commissioning, set ext addr based on factory-assigned IEEE EUI-64.

### LINKED ISSUE #573: Commissioning: Use HashMac as aExtendedAddress for commissioning process
Thread Specification 1.1 (Table 3-2, 7.4.1) requires three different kinds of Mac addresses which will be used on different phases:

- FactoryMAC address: factory assigned IEEE EUI-64;

- HashMac address : first 64 bits of the result of computing SHA-256 over FactoryMAC address (IEEE EUI-64), **used as aExtendedAddress during commissioning process**;

- RandomMac address: random generated 64-bit value, used as aExtendedAddress after commissioning succeed.

We need to follow specification to pass certification tests, since harness will verify that logic. For example, in test case 8.1.1(DUT as Joiner):
- Harness asks user to input DUT's FactoryMAC address;

- Harness calculates DUT's HashMac address by itself;

- This HashMac address will be used as src address filter to verify particular packets.

### PR DESCRIPTION
**Pull request by jwhui**
Resolves #573.

### CODE REVIEW

**Comment by chshu** [comment]
Thanks @jwhui , I will run a test with Harness and update the result.

File: src/core/meshcop/joiner.cpp [hunk 1]
```diff
@@ -65,6 +67,19 @@ Joiner::Joiner(ThreadNetif &aNetif):
...
+        uint8_t buf[Crypto::Sha256::kHashSize];
+    };
+    Crypto::Sha256 sha256;
+
+    // use extended address based on factory-assigned IEEE EUI-64
+    otPlatRadioGetIeeeEui64(NULL, buf);
+    sha256.Start();
+    sha256.Update(buf, OT_EXT_ADDRESS_SIZE);
+    sha256.Finish(buf);
+    mNetif.GetMac().SetExtAddress(extAddress);
```
Comment (chshu) [inline src/core/meshcop/joiner.cpp]: Need to update link local address here:
`mNetif.GetMle().UpdateLinkLocalAddress();`

File: src/core/meshcop/joiner.cpp [hunk 2]
```diff
@@ -296,6 +312,14 @@ void Joiner::HandleJoinerEntrust(Coap::Header &aHeader, Message &aMessage, const
 
     otLogInfoMeshCoP("join success!\r\n");
 
+    // configure a random Extended Address
+    for (size_t i = 0; i < sizeof(extAddress); i++)
+    {
+        extAddress.m8[i] = static_cast<uint8_t>(otPlatRandomGet());
+    }
+
+    mNetif.GetMac().SetExtAddress(extAddress);
```
Comment (chshu) [inline src/core/meshcop/joiner.cpp]: Need to update link local address here:
`mNetif.GetMle().UpdateLinkLocalAddress();`

File: src/core/meshcop/joiner.cpp (hunk 1 above)
Comment (jwhui) [inline src/core/meshcop/joiner.cpp]: Fixed.

**Comment by chshu** [comment]
Verified with test Harness, the issue is fixed, thanks @jwhui.

**Comment by chshu** [comment]
LGTM:+1:

**Pull request merged**